### PR TITLE
Lineup

### PR DESCRIPTION
**Pull request by jacksherling**
## Description
Adds lineup position, either 

## Associated / Resolved Issue
Resolves [ClickUp card](https://app.clickup.com/t/8677t9bb5)

### Test Case 1
1. Use penaltylineup or sidelineup enum values and view result in sim

## Review Checklist

- [ ] **Docstrings**: All methods and classes should have the file appropriate docstrings which follow the guidelines in the ["Contributing" page](https://rj-rc-software.readthedocs.io/en/latest/contributing.html) of our docs.
- [ ] **Remove extra print statements**: Any print statements used for debugging should be removed
- [ ] **Tag reviewers**: Tag some people for review and ping them on Slack

### CODE REVIEW

File: soccer/src/soccer/strategy/agent/position/their_side_lineup.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+std::optional<RobotIntent> TheirSideLineup::derived_get_task(RobotIntent intent) {
+    WorldState* world_state = this->world_state();
+    int visibleRobots = world_state->our_robots.size();
+
+    rj_geometry::Point target_pt;
+    double fieldLength = field_dimensions_.floor_length() / 2;
+    double y = field_dimensions_.their_left_goal_post_coordinate().y();
+    double x = field_dimensions_.their_left_corner().x();
+    // double x = field_dimensions_.our_right_corner().x(); // IF OTHER SIDE
```
Comment (kfu02) [inline soccer/src/soccer/strategy/agent/position/their_side_lineup.cpp]: delete

File: soccer/src/soccer/strategy/agent/position/their_side_lineup.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#include <rj_msgs/action/robot_move.hpp>
+
+#include "planning/instant.hpp"
+#include "position.hpp"
+#include "rj_geometry/point.hpp"
+
+namespace strategy {
+
+/*
+ * Vertical line on our side
```
Comment (kfu02) [inline soccer/src/soccer/strategy/agent/position/their_side_lineup.hpp]:
```suggestion
 * Vertical line on their side
```

File: soccer/src/soccer/strategy/agent/position/their_side_lineup.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    void derived_pass_ball() override{};
+    void derived_acknowledge_ball_in_transit() override{};
+
+private:
+    int r_id_{};
+
+    std::optional<RobotIntent> derived_get_task(RobotIntent intent) override;
+
+    rj_geometry::Point init_ball_pos_{};
+    bool init_ball_pos_valid_{false};
```
Comment (kfu02) [inline soccer/src/soccer/strategy/agent/position/their_side_lineup.hpp]: delete

File: soccer/src/soccer/strategy/agent/position/their_side_lineup.hpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+public:
+    TheirSideLineup(int r_id);
+    ~TheirSideLineup() override = default;
+
+    void derived_acknowledge_pass() override{};
+    void derived_pass_ball() override{};
+    void derived_acknowledge_ball_in_transit() override{};
+
+private:
+    int r_id_{};
```
Comment (kfu02) [inline soccer/src/soccer/strategy/agent/position/their_side_lineup.hpp]: figure out how to use Position superclass's r_id_ (it has this field already)

**Review by kfu02 — CHANGES_REQUESTED** [review]
Hey @jacksherling , there are some style fixes and minor things to clean up, could you handle that by next Tuesday?

**Comment by sid-parikh** [comment]
See comp fork for a working version of this (if it isn't already in ros2)

**Pull request closed without merging**